community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/schema/NativeLuceneFusionSchemaIndexProviderFactory.java

```diff
@@ -87,12 +88,14 @@ public static FusionSchemaIndexProvider newInstance( PageCache pageCache, File s
         boolean readOnly = isReadOnly( config, operationalMode );
         NumberSchemaIndexProvider nativeProvider =
                 new NumberSchemaIndexProvider( pageCache, fs, childDirectoryStructure, monitor, recoveryCleanupWorkCollector, readOnly );
+        SpatialFusionSchemaIndexProvider spatialProvider =
+                new SpatialFusionSchemaIndexProvider( pageCache, fs, childDirectoryStructure, monitor, recoveryCleanupWorkCollector, readOnly );
         LuceneSchemaIndexProvider luceneProvider = LuceneSchemaIndexProviderFactory.create( fs, childDirectoryStructure, monitor, config,
                 operationalMode );
         boolean useNativeIndex = config.get( GraphDatabaseSettings.enable_native_schema_index );
         int priority = useNativeIndex ? PRIORITY : 0;
         return new FusionSchemaIndexProvider( nativeProvider,
-                luceneProvider, new NativeSelector(), DESCRIPTOR, priority, directoriesByProvider( storeDir ), fs );
+                spatialProvider, luceneProvider, new NativeSelector(), DESCRIPTOR, priority, directoriesByProvider( storeDir ), fs );
```

Comment:
We should think this through a few times more, but based on our last discussion this does not seem to be a problem. Since the new types and new indexes are being added together, there does not exist a released version that could create a problem. For example, if we run in 3.4, add points to the database (and they are added to lucene-native and stored in the spatial index), then downgrade to 3.3 and add more points, they will not be stored in either the property store or the index (since 3.3 does not support that at all), so the spatial index remains hidden. Upgrading to 3.4 again will bring it back in a consistent state.

The same is obviously not true for a native string index, since the same type will be stored in lucene in older versions, so you have to be explicit about the type of index.

Reply:
I think you might be right about this! Let's re-iterate later.